urllib3/sync_connection.py

```diff
@@ -242,14 +243,14 @@ def consume_bytes(data):
         # re-use this connection, even though we can't. So record this in
         # h11's state machine.
         # XX need to implement this in h11
-        #state_machine.poison()
+        # state_machine.poison()
         # XX kluge for now
         state_machine._cstate.process_error(state_machine.our_role)
 
-    return _response_from_h11(h11_response)
+    return _response_from_h11(h11_response, request_bytes_iterable)
```

Comment:
Ah, here's the problem with `read` :-). (Or at least, the first problem :-).)

`request_bytes_iterable` is the body of our request, i.e. what we're uploading. The second argument to `_response_from_h11` is the body of the response, i.e. what we're downloading. The way the code is written, `SyncHTTP1Connection` actually acts as the response body (it implements the async iterator interface, see its `__anext__` method), so the second argument here should just be `self`.

Reply:
This comment was really useful.